test/servertester.go

```diff
@@ -138,6 +138,18 @@ func (st *serverTester) writeSettingsAck() {
 	}
 }
 
+func (st *serverTester) wantRSTStream() *http2.RSTStreamFrame {
```

Comment:
Pass the code expected and verify that as well?

Reply:
Added a check for the error code.